lib/Profile.dart

```diff
@@ -33,7 +34,7 @@ class _ProfileState extends State<Profile> {
         automaticallyImplyLeading: true,
         leading: IconButton(
           icon: Icon(Icons.arrow_back_ios),
-          onPressed: () => Navigator.pop(context, false),
+          onPressed: () { Home(); }
```

Comment:
I recommend using the navigator and popping. This way you can think of navigation and pushing and popping on a stack and there will not be multiple instances of the homepage running in the app. Popping from this page will return you to the homepage anyway because that is where you navigated from.